### PR TITLE
change search result approach

### LINKED ISSUE #218: Searching result
Please show result as provider and can be expanded to show it's courses whatever what category is searched

### LINKED ISSUE #219: Searching result on what to study
Show provider when searching what to study but related with the course

### PR DESCRIPTION
**Pull request by meomancer**
@cchristelis @dimasciput 
this fixes #218 , fixes #219 

fixes how result of search is represented.
although what to study is searched, still show provider but that has the query course

screenshot from 2016-09-05 13 59 07
screenshot from 2016-09-05 13 59 18
screenshot from 2016-09-05 13 59 33

### CODE REVIEW

**Comment by dimasciput** [comment]
I just tested this in my machine, the loading time for searching courses is really slow. I will try to debug and see what's going on.

**Comment by dimasciput** [comment]
@meomancer apparently the culprit is this code 

```
campuses = campuses.filter(
            courses__course_description__icontains=q
        )
```

It will return duplicate rows, you need to add distinct before filter

```
campuses = campuses.distinct().filter(
            courses__course_description__icontains=q
        )
```

File: django_project/feti/views/api.py [hunk 1]
```diff
@@ -33,18 +31,20 @@ def get(self, request, format=None):
...
             )
-        else:
-            campuses = Campus.objects.filter(
-                Q(campus__icontains=q) |
-                Q(provider__primary_institution__icontains=q)
-            )
+
+        campuses.order_by('campus')
         serializer = CampusSerializer(campuses, many=True)
+        print(serializer.data)
```
Comment (dimasciput) [inline django_project/feti/views/api.py]: I don't think we need this here

**Comment by meomancer** [comment]
@dimasciput 
PR updated
apparently because we search course first and after that we search campus that contains that course. 

Maybe for peformance, we could use pagination on scroll?

**Comment by dimasciput** [comment]
LGTM

**Pull request merged**